ctapipe/analysis/tests/test_sensitivity.py

```diff
@@ -0,0 +1,219 @@
+
+    # energy histogram for the generated events
+    energy_sim_hist_gamma = np.histogram(energy_sim_gamma,
+                                         bins=energy_edges)[0]
+    energy_sim_hist_elect = np.histogram(energy_sim_elect,
+                                         bins=energy_edges)[0]
+    energy_sim_hist_proton = np.histogram(energy_sim_proton,
+                                          bins=energy_edges)[0]
+
+    # constructer gets fed with energy and angular offset lists and desired energy binning
+    Sens = SensitivityPointSource(
+                    mc_energies={'g': energy_sel_gamma,
+                                 'p': energy_sel_proton,
+                                 'e': energy_sel_elect},
+                    energy_bin_edges={"g": energy_edges,
+                                      "p": energy_edges,
+                                      'e': energy_edges})
+
+    # effective areas
+    Sens.get_effective_areas(
+                    generator_energy_hists={'g': energy_sim_hist_gamma,
+                                            'p': energy_sim_hist_proton,
+                                            'e': energy_sim_hist_elect},
+                    generator_areas={'g': gen_area_g,
+                                     'p': gen_area_p,
+                                     'e': gen_area_g})
+
+    # midway result are the effective areas
+    eff_a = Sens.effective_areas
+    eff_area_g, eff_area_p, eff_area_e = eff_a['g'], eff_a['p'], eff_a['e']
+
+    # "selected" events are half of the "generated" events
+    # so effective areas should be half of generator areas, too
+    np.testing.assert_allclose(eff_area_g.value, gen_area_g.value/2)
+    np.testing.assert_allclose(eff_area_p.value, gen_area_p.value/2)
+
+
+def test_SensitivityPointSource_sensitivity_MC():
+
+    alpha = 1.
+    Nsig = 20
+    Nbgr = 10
+    energy_bin_edges = np.array([.1, 10])*u.TeV  # one bin with the bin-centre at 1 TeV
+    energies_sig = np.ones(Nsig)
+    energies_bgr = np.ones(Nbgr)
+
+    sens = SensitivityPointSource(
+            {'s': energies_sig*u.TeV,   # all events have 1 TeV energy
+             'b': energies_bgr*u.TeV})  # all events have 1 TeV energy
+    sens.event_weights = {'s': energies_sig, 'b': energies_bgr}
+    sensitivity = sens.get_sensitivity(alpha=alpha, signal_list=("s"), mode="MC",
+                                       sensitivity_source_flux=crab_source_rate,
+                                       sensitivity_energy_bin_edges=energy_bin_edges)
+
+    # the ratio between sensitivity and reference flux (i.e. from the crab nebula) should
+    # be the scaling factor that needs to be applied on Nsig to produce a 5 sigma result
+    # in the lima formula
+    ratio = sensitivity["Sensitivity"][0]/(crab_source_rate(1*u.TeV)).value
+    np.testing.assert_allclose([sigma_lima(ratio*Nsig+alpha*Nbgr, Nbgr, alpha)], [5])
+
+
+def test_SensitivityPointSource_sensitivity_data():
+
+    alpha = 1.
+    Non = 30
```

Comment:
Suggest to always follow the Python style guide, i.e. use `n_on` and not `Non`.
There's no need to do this via code review. E.g. [flake8](http://flake8.pycqa.org/en/latest/) will list you all of these superficial coding style violations.